src/base/ct2ctml.cpp

```diff
@@ -85,6 +89,52 @@ static std::string call_ctml_writer(const std::string& text, bool isfile)
     } else {
         file = "<string>";
         arg = "text=r'''" + text + "'''";
+
+    }
+
+    // If the user wants to convert a mechanism using a text passed via the
+    //       source="""..."""
+    // argument in python, then we have to make sure that it is short enough
+    // to fit in the command line when routed to python as:
+    //       python -c ...
+    // statement downstream in the code
+
+    // So, check the max size of a string that can be passed on the command line
+    // This is OS Specific. *nix systems have the sysconf() function that tells 
+    // us the largest argument we can pass. Since such a function does not exist
+    // for Windows, we set an arbitrary limit of 80,000 chars
+#ifdef _WIN32
+    int max_argv_size = 80000;
+#endif
+
+#ifndef _WIN32
+    int max_argv_size = sysconf(_SC_ARG_MAX);
+#endif
+
+    if (text.size() > max_argv_size) {
+
+      // If the file is too big to be passed as a command line argument later
+      // in the file, then create a temporary file and execute this function
+      // as though an input file was specified as the source. Notify the user
+      ofstream temp_cti_file("tempCtiFile.cti");
+
+      if(temp_cti_file) {
+	temp_cti_file << text;
+	file = "tempCtiFile.cti";
+	arg = "r'tempCtiFile.cti'";
+
+	writelog("Created tempCtiFile.cti\n");
+      }
+      else {
+	// If we are here, then a temp file could not be created
+	stringstream message;
```

Comment:
The `stringstream` here is unnecessary -- just supply this as the argument to `CanteraError` (you can just indent subsequent lines by one level instead of all the way to the open parentheses in cases like this). I might also recommend a more succinct error message.